packages/esbuild/src/executors/esbuild/lib/build-esbuild-options.ts

```diff
@@ -123,6 +127,142 @@ export function buildEsbuildOptions(
   return esbuildOptions;
 }
 
+/**
+ * When using TS project references we need to map the paths to the referenced projects.
+ * This is necessary because esbuild does not support project references out of the box.
+ * @param context ExecutorContext
+ */
+export function createPathsFromTsConfigReferences(
+  context: ExecutorContext
+): Record<string, string[]> {
+  const {
+    findAllProjectNodeDependencies,
+  } = require('nx/src/utils/project-graph-utils');
+  const {
+    isValidPackageJsonBuildConfig,
+  } = require('@nx/js/src/plugins/typescript/util');
+  const { readTsConfig } = require('@nx/devkit');
+  const deps = findAllProjectNodeDependencies(
+    context.projectName,
+    context.projectGraph
+  );
+  const tsConfig = readJsonFile(
+    joinPathFragments(context.root, 'tsconfig.json')
+  );
+  const referencesAsPaths = new Set(
+    tsConfig.references.reduce((acc, ref) => {
+      if (!ref.path) return acc;
+
+      const fullPath = joinPathFragments(workspaceRoot, ref.path);
+
+      try {
+        if (lstatSync(fullPath).isDirectory()) {
+          acc.push(fullPath);
+        }
+      } catch {
+        // Ignore errors (e.g., path doesn't exist)
+      }
+
+      return acc;
+    }, [])
+  );
+
+  const getTsConfigPath = (
+    projectType: string,
+    projectPath: string
+  ): string => {
+    if (projectType === 'app') {
+      return existsSync(join(projectPath, 'tsconfig.app.json'))
+        ? 'tsconfig.app.json'
+        : 'tsconfig.json';
+    }
+
+    return existsSync(join(projectPath, 'tsconfig.lib.json'))
+      ? 'tsconfig.lib.json'
+      : 'tsconfig.json';
+  };
+
+  // for each dep we check if it contains a build target
+  // we only want to add the paths for projects that do not have a build target
+  return deps.reduce((acc, dep) => {
+    const projectNode = context.projectGraph.nodes[dep];
+    const projectPath = joinPathFragments(workspaceRoot, projectNode.data.root);
+    const projTsConfig = readTsConfig(
```

Comment:
I think we can use `findRuntimeTsConfigName`, and if it returns `null` then assume that the project is non-buildable.

Reply:
`findRuntimeTsConfigName` requires `tree` but this is the executor that has context.

Reply:
`tree` which is required for `findRuntimeTsConfigName`  doesn't typical exists inside an executor and it doesn't seem like a common pattern to create it using `FsTree`.

Reply:
It could be done like `isUsingTsSolutionSetup`, where `FsTree` is used if not passed.

```
export function isUsingTsSolutionSetup(tree?: Tree): boolean {
  tree ??= new FsTree(workspaceRoot, false);

  return (
    isUsingPackageManagerWorkspaces(tree) &&
    isWorkspaceSetupWithTsSolution(tree)
  );
}
```